presto-sqlserver/src/main/java/io/prestosql/plugin/sqlserver/SqlServerClient.java

```diff
@@ -428,9 +428,10 @@ public void setNull(PreparedStatement statement, int index)
         return handle.createQuery("" +
                 "SELECT data_compression_desc FROM sys.partitions p " +
                 "INNER JOIN sys.tables t ON p.object_id = t.object_id " +
+                "INNER JOIN sys.schemas s ON t.schema_id = s.schema_id " +
                 "INNER JOIN sys.indexes i ON t.object_id = i.object_id " +
-                "WHERE SCHEMA_NAME(t.schema_id) = :schema AND t.name = :table_name AND i.type IN (0,1) " +
-                "AND i.data_space_id NOT IN (SELECT data_space_id FROM sys.partition_schemes)")
+                "WHERE s.name = :schema AND t.name = :table_name AND p.index_id = 0 AND i.type = 0 " +
```

Comment:
> `p.index_id = 0 AND i.type = 0`

why do you need this check? 

just for the info:

> sys.partitions.index_id
> Indicates the ID of the index within the object to which this partition belongs.
> 0 = heap
> 1 = clustered index
> 2 or greater = nonclustered index
> https://docs.microsoft.com/en-us/sql/relational-databases/system-catalog-views/sys-partitions-transact-sql?view=sql-server-ver15

> sys.indexes.type 
> Type of index:
> 0 = Heap
> 1 = Clustered
> 2 = Nonclustered
> 3 = XML
> 4 = Spatial
> 5 = Clustered columnstore index. Applies to: SQL Server 2014 (12.x) and later.
> 6 = Nonclustered columnstore index. Applies to: SQL Server 2012 (11.x) and later.
> 7 = Nonclustered hash index. Applies to: SQL Server 2014 (12.x) and later.
> https://docs.microsoft.com/en-us/sql/relational-databases/system-catalog-views/sys-indexes-transact-sql?view=sql-server-ver15

Reply:
Currently we have planned to support only for Heap tables. So we have restricted for other types of object

Reply:
could you please add two `/* Heap */` comments near `= 0`

Reply:
Good idea to add a comment:
```
"WHERE s.name = :schema AND t.name = :table_name " +
"AND p.index_id = 0 " + // ....
"AND i.type = 0 " // 0 means Heap index type
```

Reply:
I think comments within sql fit better here.

Reply:
@ssheikin no need to send our comments to SQL Server, is there?

Reply:
You are right. Is it expensive?

Reply:
Not at all. Just redundant.